### ISSUE TITLE
File paths

### ISSUE THREAD

**Issue by wayferer** [jmhobbs/domain-swap#17]
This is very handy! But not sure if I'm doing something wrong, is it possible to have varying paths or is that out of scope? i.e.

URLS:
some.dev.thing.com/project/folder
localhost/git/project/folder

Hope that makes sense.

**Comment by zachleat** [comment]
Yes, I think that should be in the roadmap! It is mentioned in #5

**Comment by wayferer** [comment]
Awesome. Will keep an eye out.

**Comment by zachleat** [comment]
Actually, I think we should keep this open. #5 is a bit too generic. We’ll probably move those requests into separate tickets and close it.

**Comment by wayferer** [comment]
Makes sense :smile:

**Comment by jmhobbs** [comment]
If we are mapping 1-to-1, this is pretty simple to implement.

Once we enter the realm of multiple domains it gets harder.  My current thought is that each trigger domain will have a regex that maps it's path into a canonical path, and then a map from canonical path to it's own version.

For example;

```
example.com    /^(.*)/    \1
example.local   /^git\/(.*)/  /git/\1
```

The generic mapping for example.com would be the default, I just included it for reference.

Does that make sense to anyone else?  It seems too complicated to me.  Better ideas?

**Comment by jmhobbs** [comment]
Do we anticipate that path re-maps are only going to be prefixes? If so, that would really simplify it.

I can't currently think of another case for that.

**Comment by wayferer** [comment]
I'm sure if I understand but maybe if I explain my situation it will help? basically I develop locally and have IIS setup to point to a top level folder i.e. C:\Projects then inside 'Projects' I have the following:
- git/projectName/web/(site files)
- svn/projectName/web/(site files)

When I want to show the site to others I upload the site to the preview environment:
- preview.com/projectName/web/(site files)

Sometimes it might look like
- preview.com/projectName/web-v3/(site files)

So my hope was that I could do something like:
Name: Project Name
Domains:
- preview.com/projectName/web-v3
- localhost/git/projectName/web

Then when I have localhost/git/projectName/web/index.html open I can switch it to preview.com/projectName/web-v3/index.html

Is that useful?

**Comment by jmhobbs** [comment]
@wayferer  That is useful!  I assumed that all mappings would be just prefix, but you presented a counter example.

That rules out the simple way :)

**Comment by wayferer** [comment]
Sorry :smile:

**Comment by zachleat** [comment]
Aren’t these still prefixes?
1. Search through domain set to find the URL that matches the current
2. Do a replace(CURRENT, '') to get the leftover path
3. Append to the new domain

**Comment by jmhobbs** [comment]
Except for /web-v3 vs. /web

**Comment by jmhobbs** [comment]
Oh.  Wait.  Good point.

Simple way again!

**Comment by zachleat** [comment]
Oh, I see what you’re saying:

preview.com/projectName/web-v3
preview.com/projectName/web

I think we can assume/add a trailing slash there for matching.

**Comment by wayferer** [comment]
I think so. There's no reason why I can't enter:
- preview.com/projectName/web-v3/
- localhost/git/projectName/web/

In the Domains box.

**Comment by zachleat** [comment]
I wonder if it’d be easier in the long run to use something like http://medialize.github.io/URI.js/

Just an idea.

**Comment by jmhobbs** [comment]
I added a feature branch for this, and have a working version with limited testing.

Relevant commits: a3061ee0f74d0e7c4279be70b1e06d5024721142 and 3bccb39d5338492b0e74b0445c3fda782efc2c59

Basically, I just made it so that you enter them with an optional path that will get added/removed when changing to/from that domain.

![input](https://dl.dropbox.com/u/28665584/screenshots/2013-12-12_15:57:09_201x202_xaijeafief.jpg)

This is pretty quick and dirty.  It needs to be better tested (I'm sick of qUnit btw) and a migration path should be figured out since the domain set storage is a different format.

**Comment by wayferer** [comment]
I'm happy to test if you can give me a link to a build?

**Comment by jmhobbs** [comment]
Sure thing.

https://dl.dropboxusercontent.com/u/28665584/file-drop/google-chrome-domain-swap-3bccb39.crx

**Comment by wayferer** [comment]
ok the icon shows up now (didn't before) but clicking it doesn't do anything?

**Comment by jmhobbs** [comment]
Check the options page to make sure your sets are in there, and then re-save from that page.

Might be an upgrade bug, I didn't work on any of that logic at all.

**Comment by wayferer** [comment]
I uninstalled the previous one. When I open the options page my set is there.

**Comment by jmhobbs** [comment]
Hmmmm....

Can you paste in your set so I have something to work with?  You can sanitize as needed to protect the innocent, but try to stay true to the data.

**Comment by wayferer** [comment]
Sure :)
- url.to.previewserv.er/Project/m/
- localhost/svn/Project/web/

Is that would you need?
